### PR TITLE
Devel 4.0: CI test

### PR DESCRIPTION
**Pull request by vrutkovs**
Merge `devel-40` changes in master

TODO:

* GCP:
  * [x] the secret doesn't contain `pull-secret`. https://jira.coreos.com/browse/DPTP-278
  * [x] `openshift_node_port_range` is undefined
  * [x] Can't connect to build image
  * [x] `'openshift_deployment_type' is undefined`
  * [x] Wrong base domain specified, https://github.com/openshift/release/pull/2158, https://github.com/openshift/release/pull/2159
  * [x] Scale bootstrap instance group to 0
  * [x] Update public API DNS entry to exclude bootstrap
        Teardown needs to be fixed too 
  * [x] MCD notices a change from Google in `/etc/hosts` and reboots nodes
  * [x] Fix public API DNS update - bootstrap is still mentioned
  * [ ] Create etcd discovery entry via ansible task
  * [x] Wait for ingress controller to appear
  * [x] Adjust MTU on server - 1500 is correct
  * [ ] Wait for all first level operators have completed
  * [ ] Ensure test suite is running and passes
  * [ ] Rewrite tasks using `k8s` and `k8s_facts` module instead of `oc_obj`
  * [ ] Rework worker join: start control-plane first and setup workers (incl. kubelet and crio) before bootkube finishes

### CODE REVIEW

File: playbooks/deploy_cluster_40.yml [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+      name: container_runtime
+      tasks_from: docker_storage_setup_overlay.yml
+  - import_role:
+      name: container_runtime
+      tasks_from: extra_storage_setup.yml
+  - import_role:
+      name: container_runtime
+      tasks_from: package_crio.yml
+  - name: FIXME pause_image
+    ini_file:
```
Comment (michaelgugino) [inline playbooks/deploy_cluster_40.yml]: We shouldn't need to do this.

Comment (vrutkovs) [reply playbooks/deploy_cluster_40.yml]: Currently it sets `docker.io/openshift/origin-pod:${version}` - seem oreg_url formatting is broken

Comment (vrutkovs) [reply playbooks/deploy_cluster_40.yml]: Right, its no longer required, removed that part

**Comment by sdodson** [comment]
This is the first step towards 4.0 BYOH. There will likely be many followup changes.

**Pull request merged**